packages/components/video/VideoCard.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState } from "react";
+import {
+  StyleProp,
+  StyleSheet,
+  TextStyle,
+  View,
+  ViewStyle,
+} from "react-native";
+
+import DefaultVideoImage from "../../../assets/default-images/default-video-thumbnail.png";
+import { Post } from "../../api/feed/v1/feed";
+import { useNSUserInfo } from "../../hooks/useNSUserInfo";
+import { parseUserId } from "../../networks";
+import { ipfsURLToHTTPURL } from "../../utils/ipfs";
+import { prettyMediaDuration } from "../../utils/mediaPlayer";
+import { zodTryParseJSON } from "../../utils/sanitize";
+import { neutral22, neutral77, withAlpha } from "../../utils/style/colors";
+import { fontSemibold13, fontSemibold14 } from "../../utils/style/fonts";
+import { layout } from "../../utils/style/layout";
+import { tinyAddress } from "../../utils/text";
+import { BrandText } from "../BrandText";
+import { OmniLink } from "../OmniLink";
+import { OptimizedImage } from "../OptimizedImage";
+import { CustomPressable } from "../buttons/CustomPressable";
+import { UserAvatarWithFrame } from "../images/AvatarWithFrame";
+import { ZodSocialFeedVideoMetadata } from "../socialFeed/NewsFeed/NewsFeed.type";
+import { DateTime } from "../socialFeed/SocialCard/DateTime";
+import { SpacerColumn, SpacerRow } from "../spacer";
+
+const IMAGE_HEIGHT = 173;
+export const VIDEO_CARD_WIDTH = 261;
+export const VideoCard: React.FC<{
+  post: Post;
+  hideAuthor?: boolean;
+  style?: StyleProp<ViewStyle>;
+}> = ({ post, hideAuthor, style }) => {
+  const video = zodTryParseJSON(ZodSocialFeedVideoMetadata, post.metadata);
+  const authorNSInfo = useNSUserInfo(post.authorId);
+  const [, userAddress] = parseUserId(post.authorId);
+  const [isHovered, setIsHovered] = useState(false);
+
+  let cardWidth = StyleSheet.flatten(style)?.width;
+  if (typeof cardWidth !== "number") {
+    cardWidth = VIDEO_CARD_WIDTH;
+  }
+
+  const username = authorNSInfo?.metadata?.tokenId
+    ? authorNSInfo?.metadata?.tokenId
+    : tinyAddress(userAddress, 16);
+
+  if (post.identifier.startsWith("padded-")) {
+    return <View style={{ width: cardWidth, height: 381 }} />;
+  }
+
+  if (!video) return null;
+  return (
+    <View style={unitCardStyle}>
+      <CustomPressable
+        style={imgBoxStyle}
+        onHoverIn={() => setIsHovered(true)}
+        onHoverOut={() => setIsHovered(false)}
+        // TODO:
+        // onPress={() => {
```

Comment:
is this todo forgotten?

Reply:
No, FeedVideoViewScreen will be made in another PR